components/camel-rest-swagger/src/main/java/org/apache/camel/component/rest/swagger/RestSwaggerEndpoint.java

```diff
@@ -0,0 +1,313 @@
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.util.LoadPropertiesException;
+import org.apache.camel.util.ObjectHelper;
+import org.apache.camel.util.StringHelper;
+
+import io.swagger.models.HttpMethod;
+import io.swagger.models.Operation;
+import io.swagger.models.Path;
+import io.swagger.models.Swagger;
+import io.swagger.parser.SwaggerParser;
+import io.swagger.util.Json;
+
+/**
+ * An awesome REST endpoint backed by Swagger specifications.
+ */
+@UriEndpoint(firstVersion = "2.19.0", scheme = "rest-swagger", title = "REST Swagger",
+             syntax = "rest-swagger:operationId:specificationPath", label = "rest,swagger,http", producerOnly = true)
+public final class RestSwaggerEndpoint extends DefaultEndpoint {
+
+    private static final String DEFAULT_SPECIFICATION_PATH = "swagger.json";
+
+    static final String[] KNOWN_WORKING_COMPONENTS = new String[] {"http", "http4", "netty4-http", "restlet", "jetty",
+            "undertow"};
+
+    @UriParam(description = "REST producer component to use, for instance `undertow`, if not defined it is expected that only one REST producer Camel component will be present on the classpath.",
+              label = "producer")
+    private String componentName;
+
+    @UriParam(description = "Scheme, hostname and optionally port to use, in the form of `http[s]://hostname[:port]`",
+              label = "producer")
+    private String endpoint;
+
+    @UriPath(description = "ID of the operation from the Swagger specification", label = "producer")
+    @Metadata(required = "true")
+    private String operationId;
+
+    @UriPath(description = "Path to the Swagger specification file", defaultValue = DEFAULT_SPECIFICATION_PATH,
+             defaultValueNote = "By default loads `swagger.json` file")
+    private String specificationPath = DEFAULT_SPECIFICATION_PATH;
+
+    public RestSwaggerEndpoint() {
+    }
+
+    public RestSwaggerEndpoint(final String uri, final String remaining, final RestSwaggerComponent component) {
+        super(uri, component);
+
+        operationId = Optional.ofNullable(StringHelper.before(remaining, ":")).orElse(remaining);
+
+        specificationPath = Optional.ofNullable(StringHelper.after(remaining, ":")).orElse(DEFAULT_SPECIFICATION_PATH);
+
+        setExchangePattern(ExchangePattern.InOut);
+    }
+
+    /**
+     * Loads the Swagger definition model from the given path. Tries to resolve the resource using Camel's resource
+     * loading support, if it fails uses Swagger's resource loading support instead.
+     *
+     * @param path
+     * @return
+     * @throws IOException
+     */
+    static Swagger loadSpecificationFrom(final String path) throws IOException {
+        final ObjectMapper mapper = Json.mapper();
+
+        final SwaggerParser swaggerParser = new SwaggerParser();
+
+        Swagger answer = null;
+        try (InputStream stream = ObjectHelper.loadResourceAsStream(path)) {
```

Comment:
You may need to use the one that takes CamelContext or ClassLoader (cant remember) but those support osgi and other "app servers"